src/Components/PostComment.jsx

```diff
@@ -0,0 +1,46 @@
+import { useContext, useState } from "react";
+import { postComment } from "../api";
+import { UserContext } from "./UserContext ";
+
+const PostComment = ({ article_id, setNewComment }) => {
+    const [newItem, setNewItem] = useState('')
+    const [submitActive, setSubmitActive] = useState(false); 
+    const [message, setMessage] = useState(''); 
+    const { user } = useContext(UserContext); 
+
+    const handleSubmit = (event) => {
+        event.preventDefault()
+        setSubmitActive(true); 
+        postComment(article_id, user.username, newItem)
+        .then(({ data }) => {
+            setNewComment(data);
```

Comment:
This function isn't _quite_ there yet. What we need is for the new comment to be optimistically rendered in the app at the _top_ of the list. This is what I mean...
<img width="488" alt="Screenshot 2022-08-25 at 09 16 46" src="https://user-images.githubusercontent.com/67956822/186612874-e5d46a1a-e630-41a3-bcc4-2bb01e71daab.png">
 
So we optimistically rerender the list, with the new comment at the top with some fake data so that it can be added to the list, and THEN the API call happens. It's a really small distinction but quite important.